### PR TITLE
[E2E] Create stableswap and cl pools

### PR DESCRIPTION
**Pull request by phamminh0811**
Closes: #3742 

## What is the purpose of the change

This pull request add CreateStableSwapPool function to e2e test and also create stableswap and CL pools to setup test

## Testing and Verifying

This change is already covered by existing tests.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: tests/e2e/configurer/chain/commands.go
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:37]:
```suggestion
```

File: tests/e2e/configurer/upgrade.go
Comment (p0mvn) [inline tests/e2e/configurer/upgrade.go:26]: Why this change?

File: tests/e2e/configurer/upgrade.go
Comment (p0mvn) [inline tests/e2e/configurer/upgrade.go:139]: Actually, I think we should hold-off with pre-creating CL pools until after concentrated liquidity is released. The reason is there are no CL pools existing in the pre-upgrade version

File: tests/e2e/configurer/upgrade.go
Comment (phamminh0811) [reply tests/e2e/configurer/upgrade.go:131]: The reason is the validator has no uusdc and milliusdc in the pre-upgrade version. So I just check the version is update to make sure e2e test work well when upgrading

Comment (p0mvn) [reply tests/e2e/configurer/upgrade.go:131]: I suggest using the denoms that already exist. e.g. "uosmo" and "stake". It doesn't matter for correctness that these denoms aren't stable in reality

File: tests/e2e/scripts/stablePool.json
Comment (RusAkh) [inline tests/e2e/scripts/stablePool.json:4]: Hello @phamminh0811. Is it possible to do it like this?
```suggestion
	"exit-fee": "0",
```

**Comment by RusAkh** [comment]
There is no need to commit `rate_limiter.wasm`, this file is generated when running e2e, @phamminh0811, could you please remove it?

File: tests/e2e/scripts/stablePool.json
Comment (RusAkh) [inline tests/e2e/scripts/stablePool.json:4]:
```suggestion
	"exit-fee": "0",
```

**Review by RusAkh — CHANGES_REQUESTED** [review]
LGTM when comments addressed. Please, reset `rate_limiter.wasm` to the commit of development:

```
git reset <latest main commit hash> -- <path to rate limiter>
```

**Comment by phamminh0811** [comment]
> LGTM when comments addressed. Please, reset `rate_limiter.wasm` to the commit of development:
> 
> ```
> git checkout <latest main commit hash> -- <path to rate limiter>
> ```

seems like this wasm file is deleted from main branch since this commit #4208 
should we reset them?

**Comment by RusAkh** [comment]
> > LGTM when comments addressed. Please, reset `rate_limiter.wasm` to the commit of development:
> > ```
> > git checkout <latest main commit hash> -- <path to rate limiter>
> > ```
> 
> seems like this wasm file is deleted from main branch since this commit #4208 should we reset them?

I see, I think then you may merge main to this PR and the issue is solved! Lmk when this is ready!

File: tests/e2e/scripts/stablePool.json
Comment (pysel) [reply tests/e2e/scripts/stablePool.json:4]: hey @phamminh0811, I think this is not resolved yet

File: tests/e2e/configurer/chain/commands.go
Comment (pysel) [inline tests/e2e/configurer/chain/commands.go:33]: nit: 
```suggestion
func (n *NodeConfig) CreateStableswapPool(poolFile, from string) uint64 {
```

**Review by pysel — CHANGES_REQUESTED** [review]

**Review by pysel — CHANGES_REQUESTED** [review]
LGTM! The only thing is `rate_limiter.wasm` file. Can you please pull changes from main, I think it should be gone then?

**Review by pysel — APPROVED** [review]

File: tests/e2e/scripts/stablePool.json
Comment (pysel) [inline tests/e2e/scripts/stablePool.json:6]: Hey @phamminh0811, could you please remove redundant whitespaces?

**Review by pysel — CHANGES_REQUESTED** [review]

File: tests/e2e/scripts/stablePool.json
Comment (phamminh0811) [inline tests/e2e/scripts/stablePool.json:6]:
```suggestion
        "scaling-factors": "1000,1"
```

File: tests/e2e/scripts/stablePool.json
Comment (pysel) [inline tests/e2e/scripts/stablePool.json:2]: nit: I think it should be all one TAB back. Correct me if I am wrong please

Comment (phamminh0811) [reply tests/e2e/scripts/stablePool.json:2]: I saw that every json file in scripts folder is the same as this file, so i think it's no need to all TAB back

Comment (pysel) [reply tests/e2e/scripts/stablePool.json:2]: got it, it just seems like there is double tab on every line

**Review by pysel — APPROVED** [review]

File: tests/e2e/configurer/upgrade.go
Comment (p0mvn) [inline tests/e2e/configurer/upgrade.go:130]: Let's also add some logic to execute a swap against one of these pools please

It could look similar to: `TestGeometricTwapMigration`

File: tests/e2e/configurer/upgrade.go
Comment (p0mvn) [inline tests/e2e/configurer/upgrade.go:144]: We should also have a separate test fixture for swapping against a stableswap pool in `e2e_test.go` (post-upgrade)

Comment (phamminh0811) [reply tests/e2e/configurer/upgrade.go:144]: Got it

Comment (p0mvn) [reply tests/e2e/configurer/upgrade.go:144]: Hii. Any updates here?

Comment (phamminh0811) [reply tests/e2e/configurer/upgrade.go:145]: @p0mvn I added new test on e2e_test file. please check

File: tests/e2e/e2e_test.go
Comment (p0mvn) [inline tests/e2e/e2e_test.go:365]: Why do we need any of these? We just want to swap against a pool that was created pre-upgrade

File: tests/e2e/e2e_test.go
Comment (p0mvn) [inline tests/e2e/e2e_test.go:345]: Here, a new pool is created but we want to swap against the pool that was created pre-upgrade

File: tests/e2e/configurer/config/constants.go
Comment (p0mvn) [inline tests/e2e/configurer/config/constants.go:40]: This has changed since #4384 

Let's make sure that we are not hard-coding this value. Instead, let's get the resulting pool id from `CreateStableSwapPool` and persist it until we reach `TestStableSwapPostUpgrade`

Can you please make a similar update for `PreUpgradePoolId`?

Comment (p0mvn) [reply tests/e2e/configurer/config/constants.go:40]: We do something similar with wallet creation where we set it pre-upgrade here:
https://github.com/osmosis-labs/osmosis/blob/7158199261ef7f5be709cd763377e4ecc2effa4d/tests/e2e/configurer/upgrade.go#L159-L163

and then use here in tests:
https://github.com/osmosis-labs/osmosis/blob/7158199261ef7f5be709cd763377e4ecc2effa4d/tests/e2e/e2e_test.go#L1116

We should have similar logic for pool ids because now we create pools with ids 1 to 833 in genesis. So your stabelswap pool number is likely to be 835 or so

**Review by p0mvn — APPROVED** [review]
LGTM nice work

**Comment by p0mvn** [comment]
Hi @phamminh0811 . Can you please resolve conflicts here? Let's get this in

**Comment by phamminh0811** [comment]
hi @p0mvn , I have resolved those conflicts, please check it

**Review by p0mvn — APPROVED** [review]
Thank you

File: go.work.sum
Comment (p0mvn) [inline go.work.sum:10]: Actually, can you please remove this file before we merge this?

**Review by p0mvn — CHANGES_REQUESTED** [review]

**Review by p0mvn — APPROVED** [review]
LGTM thank you

**Pull request merged**
